pandas/tests/computation/test_eval.py

```diff
@@ -608,6 +608,12 @@ def test_unary_in_array(self):
                       -False, False, ~False, +False,
                       -37, 37, ~37, +37], dtype=np.object_))
 
+    def test_float_comparison_bin_op(self):
+        # GH 16363
+        df = pd.DataFrame({'x': np.array([0], dtype=np.float32)})
```

Comment:
Is this specific to float32? Read briefly through issue and seemed like it affected other things

Reply:
@WillAyd The bug arises when one side returns a float32 and the other side is a scalar that doesn't have a 'value' attribute.

Reply:
Does this still work if the operands are switched? i.e. `-0.1 > x`?

Reply:
an you parameterize on the dtype (at least float32/float64)

Reply:
> Does this still work if the operands are switched? i.e. `-0.1 > x`?

I will add a test with a negative value on the left.

Reply:
> an you parameterize on the dtype (at least float32/float64)

Would you like a test for float64? This particular bug was caused by code that handles float32's, so float64's were fine, but I can add a test to ensure that this does not become an issue.

Reply:
@jreback Oh, I see now pytest's `parametrize` in some of the other tests, I will update this